atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/transaction/Transactions3TableInteraction.java

```diff
@@ -0,0 +1,154 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.backup.transaction;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.TableMetadata;
+import com.datastax.driver.core.policies.DefaultRetryPolicy;
+import com.datastax.driver.core.policies.RetryPolicy;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.utils.Bytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraConstants;
+import com.palantir.atlasdb.pue.KvsConsensusForgettingStore;
+import com.palantir.atlasdb.transaction.encoding.TwoPhaseEncodingStrategy;
+import com.palantir.atlasdb.transaction.impl.TransactionConstants;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Transactions3TableInteraction implements TransactionsTableInteraction {
+    private final FullyBoundedTimestampRange timestampRange;
+    private final RetryPolicy abortRetryPolicy;
+
+    public Transactions3TableInteraction(FullyBoundedTimestampRange timestampRange, RetryPolicy abortRetryPolicy) {
+        this.timestampRange = timestampRange;
+        this.abortRetryPolicy = abortRetryPolicy;
+    }
+
+    @Override
+    public FullyBoundedTimestampRange getTimestampRange() {
+        return timestampRange;
+    }
+
+    @Override
+    public String getTransactionsTableName() {
+        return TransactionConstants.TRANSACTIONS2_TABLE.getTableName();
+    }
+
+    @Override
+    public PreparedStatement prepareAbortStatement(TableMetadata transactionsTable, Session session) {
+        // we are declaring bankruptcy if this fails anyway
+        ByteBuffer abortCommitTsBb = ByteBuffer.wrap(TwoPhaseEncodingStrategy.ABORTED_TRANSACTION_COMMITTED_VALUE);
+
+        Statement abortStatement = QueryBuilder.update(transactionsTable)
+                .with(QueryBuilder.set(CassandraConstants.VALUE, abortCommitTsBb))
+                .where(QueryBuilder.eq(CassandraConstants.ROW, QueryBuilder.bindMarker()))
+                .and(QueryBuilder.eq(CassandraConstants.COLUMN, QueryBuilder.bindMarker()))
+                .and(QueryBuilder.eq(CassandraConstants.TIMESTAMP, CassandraConstants.ENCODED_CAS_TABLE_TIMESTAMP));
```

Comment:
Afraid I don't think this is safe. This basically just writes an abort, which is not okay if a non-abortive STAGING or COMMITTED value is present.